### ISSUE TITLE
stop generating empty spans in .coma

### ISSUE THREAD

**Issue by Armael** [creusot-rs/creusot#1157]
```
# find creusot/tests -name \*.coma -exec grep -E -e 'let%span [a-z0-9]* = ""' {} \;
...
  let%span span41 = "" 0 0 0 0
  let%span span43 = "" 0 0 0 0
...
```

why3 and why3ide complain on those spans and then emit a lot of warnings, which are hard to then distinguish from actually incorrect spans.

**Comment by xldenis** [comment]
This would be easy to fix were it not for:

```
        // If we ask for relative paths and the paths comes from the standard library, then we prefer returning
        // None, since the relative path of the stdlib is not stable.
        let path = match (&self.opts.span_mode, path) {
            (SpanMode::Relative(_), RealFileName::Remapped { .. }) => return None,
            _ => path.local_path_if_available(),
        };
```

Basically, during translation we give names to all non-dummy spans, but afterwards when we actually emit the code we only emit valid spans for non remapped paths in relative mode (aka ci). Since we can have an empty `let%span` this means we must provide a dummy value. 

The fix I can think of right now is for `Dependencies` to know the span mode and also check for this side-condition.

**Comment by Armael** [comment]
Fixed by #1161 .
